src/io/flutter/sdk/FlutterSdk.java

```diff
@@ -32,13 +38,21 @@
 import io.flutter.console.FlutterConsoleHelper;
 import io.flutter.dart.DartPlugin;
 import io.flutter.utils.FlutterModuleUtils;
+import org.jdom.Element;
+import org.jdom.JDOMException;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import java.io.File;
+import java.io.IOException;
 import java.util.concurrent.atomic.AtomicBoolean;
 
 public class FlutterSdk {
   public static final String FLUTTER_SDK_GLOBAL_LIB_NAME = "Flutter SDK";
+
+  private static final String SDK_LIBRARY_PATH = ".idea/libraries/Dart_SDK.xml";
```

Comment:
It's a callback, so there are many more things could be said:

- The basics: when is this callback invoked? (What is starting up? What is the state of the Project when it's called? What runs before and after?)
 - What is blocked from running until this callback finishes? What can be running at the same time?
 - If the action is lengthy, how is progress shown?
 - If there are multiple startup activities, what order do they run?

In this case it's fairly easy to read the source code. In other cases the calling source code is complicated. Getting the details about a callback's environment often requires coming up with experiments and using the debugger.

The reference to DumbAware is itself a somewhat cryptic clue. Looking at StartupManagerImpl, it significantly changes the behavior.

(Again, your advice is very helpful - but it could help more people if recorded in javadoc.)